service module: use sysrc on FreeBSD

problame opened the pull request:
sysrc(8) is the designated userland program to edit rc files on FreeBSD.

It first appeared in FreeBSD 9.2, hence is available on all supported
versions of FreeBSD.

Side effect: fixes #2664

problame committed:
service module: use sysrc on FreeBSD

sysrc(8) is the designated userland program to edit rc files on FreeBSD.

It first appeared in FreeBSD 9.2, hence is available on all supported
versions of FreeBSD.

Side effect: fixes #2664

gregdek commented:
Thanks @problame. To the current maintainers, @grossws please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

problame commented:
@gregdek So far, nobody has reviewed the patch. What is the normal response time to patches of this (small) size?

grossws commented:
@problame, usual time for _core_review_ was 1-3 _months_ even for bugfix patches. I volunteered to make reviews and am trying to do reviews in timeframe of a few weeks.

Patch size is not the only factor that affects review time. Presence of spare time is other important factor and familiarity with target OS is another one. To reduce review time you may provide test FreeBSD VM and/or provide some additional info like actual FreeBSD version distribution among users.

Be patient, we're not a paid support team but a community volunteers.

bcoca commented on system/service.py:
```diff
@@ -941,6 +941,7 @@ class FreeBsdService(Service):
 
     platform = 'FreeBSD'
     distribution = None
+    sysrc_cmd = "/usr/sbin/sysrc"
```
I would use the module's find_bin function to locate sysrc as you cannot always trust it to be in the default location

Another point is that this is new in v9 (iirc) so this will break the module for those using v8 and below. I know they are ancient and unsupported ... but people still run these systems. So i would fallback to the current system if sysrc is not available.

And this is just an informative note, I'm trying to deprecate this module in favor of 'per system' ones, so I might end up pulling these changes into a bsdrc module. But systemctl is only one I've had time to separate.

problame committed:
Incorporate changes suggested by bcoca.

- Use `get_bin_path` to find sysrc binary.
- Only use sysrc when available (support for legacy versions of FreeBSD)

problame commented:
I incorporated @bcoca's suggestions in the last commit.

@grossws As for the test VMs, I found the following Vagrant boxes to be quite usable (after `vagrant mutate`ing them to libvirt)

(I ran my non-automated tests against these boxes)

**FreeBSD 8.4 (no sysrc, should use legacy code path**
http://dlang.dawg.eu/vagrant/FreeBSD-8.4-i386.box
http://dlang.dawg.eu/vagrant/FreeBSD-8.4-amd64.box

**FreeBSD 9.2 (earliest sysrc available, should use new code path)**
http://iris.hosting.lv/freebsd-9.2-i386.box

**FreeBSD 10.2 (sysrc available, should use new code path)**
https://petar.xyz/files/freebsd-10.2-amd64-wunki.box

jbergstroem commented:
I think this broke new services. See https://github.com/ansible/ansible/issues/17892.
